pandera/strategies.py

```diff
@@ -213,9 +213,9 @@ def _wrapper(cls, *args, **kwargs):
 # pylint: disable=line-too-long
 # Values taken from
 # https://hypothesis.readthedocs.io/en/latest/_modules/hypothesis/extra/numpy.html#from_dtype  # noqa
-# NOTE: We're reducing the range here by an order of magnitude to avoid overflows
-# when synthesizing timezone-aware timestamps.
-MIN_DT_VALUE = -(2**62) + 1
+
+# If the datetime value is negative, then that results in `pd.NaT` in `_datetime_strategy`
+MIN_DT_VALUE = 0
```

Comment:
please revert this... it's not relevant to the double coercion issue and is out of scope for this PR.

CI hasn't had issues with this (tried this out locally and it works fine)

```
In [1]: pd.Timestamp(-(2**62) + 1)
Out[1]: Timestamp('1823-11-12 00:06:21.572612097')
```

If this continues to be an issue for you please create a new ticket and we can discuss there